.github/workflows/check-formatting.yml

```diff
@@ -0,0 +1,42 @@
+name: Format Check
+
+on:
+  push:
+    branches: master
+    paths:
+      - '**.cpp'
+      - '**.hpp'
+      - '**.c'
+      - '**.h'
+      - '**/check-formatting.yml'
+  pull_request:
+    branches: master
+    paths:
+      - '**.cpp'
+      - '**.hpp'
+      - '**.c'
+      - '**.h'
+      - '**/check-formatting.yml'
+
+jobs:
+  format-check:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v4
+
+    - name: Set up Clang Format
+      run: sudo apt-get install -y clang-format
```

Comment:
just curious, GH Actions supports caching, wonder if it's possible to cache the runner? or does it already?

Reply:
I've implemented that for the Windows workflow. I'm not sure yet where else it would be appropriate.

Reply:
> I've implemented that for the Windows workflow. I'm not sure yet where else it would be appropriate.

https://github.com/netpanzer/netpanzer/pull/133/files#diff-5ce5c9ff86f58b1f87b35b5227b2e84cb69f022d6741e1854f3e1e181091773c